subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ProblemFactory.kt

```diff
@@ -47,6 +48,11 @@ interface ProblemFactory {
          */
         fun exception(message: String): Builder
 
+        /**
+         * Uses a provided exception for this problem.
+         */
+        fun exception(exception: InvalidUserCodeException): Builder
```

Comment:
🤔 How is this related to the workaround?

Reply:
Ouch, this is a leftover from my first attempt. Removed, thank you for the review